include/grpcpp/impl/codegen/status.h

```diff
@@ -88,19 +90,21 @@ class Status {
         "Mismatched status code");
   }
 
+  explicit Status(absl::Status&& status)
+      : status_(std::forward<absl::Status>(status)) {}
+
   /// Construct an instance with associated \a code and \a error_message.
   /// It is an error to construct an OK status with non-empty \a error_message.
   Status(StatusCode code, const std::string& error_message)
-      : code_(code), error_message_(error_message) {}
+      : status_(static_cast<absl::StatusCode>(code), error_message) {}
```

Comment:
Should we add some static assertions that the individual codes in our `StatusCode` enum match the ones in `absl::StatusCode`?